### DISCUSSION TITLE
Cache expiration issue?

### DISCUSSION THREAD

**Discussion by larsparendt** [onevcat/Kingfisher#2071] [Q&A] [answered]
Is it expected behaviour that expiration for an image is postponed every time it's requested?

Example:

- I set disk and memory expiration to 5 seconds.
- The first time I want to retrieve an image (from a URL) it is naturally fetched and cached.
- However, if I keep retrieving the image BEFORE the 5 second expiration deadline is reached, it seems like the deadline keeps getting postponed and so never expires and refetches.

Is this intended behaviour?

My real life problem is that I have images that rarely change, so the default expiration time of 7 days seems perfect. However, because the image is retrieved (from Kingfisher) on a daily basis it seems like the image never expires, which causes a potential updated image (on the server) to never get fetched - unless the user doesn't use the app for more than 7 days.

Thanks.

**Comment by onevcat** [comment, marked as answer]
> Is it expected behaviour that expiration for an image is postponed every time it's requested?

Yes. It is intended. Kingfisher goes to a "refresh when read" way when determine a disk cache should be expired or not.

To change this behavior, try `.diskCacheAccessExtending`, such as:

```swift
KF.url(url)
  .diskCacheAccessExtending(.none)
  .set(to: imageView)

// Or this if you prefer the classical way
imageView.kf.setImage(with: url, options: [.diskCacheAccessExtendingExpiration(.none)])
```

For its reference, check it [here](https://swiftpackageindex.com/onevcat/kingfisher/master/documentation/kingfisher/kingfisheroptionsinfoitem/diskcacheaccessextendingexpiration(_:)) if you want to know more.

**Reply by larsparendt** [reply]
Thank you so much. Should have looked harder at the documentation, but thanks for taking the time to answer.
